src/lib/serviceRoutines/statisticsTreat.cpp

```diff
@@ -394,101 +206,158 @@ std::string statisticsTreat
-    out += TAG_ADD_COUNTER("subCacheEntries", noOfSubCacheEntries);
+    resetStatistics();
+    js.addString("message", "All statistics counter reset");
+    return js.str();
   }
 
-  if (noOfSubCacheLookups != -1)
+  /* Conditional statistics (in the same order as described in statistics.md, although
+   * beautifier tools may change this order */
+  if (countersStatistics)
   {
-    out += TAG_ADD_COUNTER("subCacheLookups", noOfSubCacheLookups);
+    js.addRaw("counters", renderCounterStats());
   }
-
-  if (noOfSubCacheRemovals != -1)
+  if (semWaitStatistics)
   {
-    out += TAG_ADD_COUNTER("subCacheRemovals", noOfSubCacheRemovals);
+    js.addRaw("semWait", renderSemWaitStats());
   }
-
-  if (noOfSubCacheRemovalFailures != -1)
+  if (timingStatistics)
   {
-    out += TAG_ADD_COUNTER("subCacheRemovalFailures", noOfSubCacheRemovalFailures);
+    js.addRaw("timing", renderTimingStatistics());
   }
-
-
-  if (semTimeStatistics)
+  if ((notifQueueStatistics) && (strcmp(notificationMode, "threadpool") == 0))
   {
-    char requestSemaphoreWaitingTime[64];
-    semTimeReqGet(requestSemaphoreWaitingTime, sizeof(requestSemaphoreWaitingTime));
-    out += TAG_ADD_STRING("requestSemaphoreWaitingTime", requestSemaphoreWaitingTime);
+    js.addRaw("notifQueue", renderNotifQueueStats());
+  }
 
-    char mongoPoolSemaphoreWaitingTime[64];
-    mongoPoolConnectionSemWaitingTimeGet(mongoPoolSemaphoreWaitingTime, sizeof(mongoPoolSemaphoreWaitingTime));
-    out += TAG_ADD_STRING("dbConnectionPoolWaitingTime", mongoPoolSemaphoreWaitingTime);
+  // Unconditional stats
+  int now = getCurrentTime();
+  js.addNumber("uptime_in_secs", now - startTime);
+  js.addNumber("measuring_interval_in_secs", now - statisticsTime);
 
-    char transSemaphoreWaitingTime[64];
-    semTimeTransGet(transSemaphoreWaitingTime, sizeof(transSemaphoreWaitingTime));
-    out += TAG_ADD_STRING("transactionSemaphoreWaitingTime", transSemaphoreWaitingTime);
+  // Special case: simulated notifications
+  int nSimNotif = __sync_fetch_and_add(&noOfSimulatedNotifications, 0);
+  renderUsedCounter(&js, "simulatedNotificaitons", nSimNotif);
 
-    char cacheSemaphoreWaitingTime[64];
-    semTimeCacheGet(cacheSemaphoreWaitingTime, sizeof(cacheSemaphoreWaitingTime));
-    out += TAG_ADD_STRING("subCacheSemaphoreWaitingTime", cacheSemaphoreWaitingTime);
+  ciP->httpStatusCode = SccOk;
+  return js.str();
 
-    char ccMutexWaitingTime[64];
-    mutexTimeCCGet(ccMutexWaitingTime, sizeof(ccMutexWaitingTime));
-    out += TAG_ADD_STRING("curlContextMutexWaitingTime", ccMutexWaitingTime);
+}
 
-    char timeStatSemaphoreWaitingTime[64];
-    semTimeTimeStatGet(timeStatSemaphoreWaitingTime, sizeof(timeStatSemaphoreWaitingTime));
-    out += TAG_ADD_STRING("timeStatSemaphoreWaitingTime", timeStatSemaphoreWaitingTime);
+/* ****************************************************************************
+*
+* statisticsCacheTreat -
+*
+* FIXME P3: maybe this function should be in the cache/ library?
```

Comment:
The "other parts" used by this function (mongoSubCacheStatisticsReset() and mongoSubCacheStatisticsGet() are in the cache library, so I guess that is ok as it is now. I'll remove the FIXME message.